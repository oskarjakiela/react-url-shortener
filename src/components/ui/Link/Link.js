import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import './Link.css';


const displayName = 'Shorty-Link';

const Link = ({
  children,
  className,
  ...props
}) => (
  <a className={classnames(displayName, className)} {...props}>
    {children}
  </a>
);

Link.displayName = displayName;

Link.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Link;
