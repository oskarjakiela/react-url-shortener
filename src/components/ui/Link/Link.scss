.Shorty-Link {
  display: inline-block;
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1rem;
  text-decoration: none;
  color: #eb4a42;
  cursor: pointer;
}

