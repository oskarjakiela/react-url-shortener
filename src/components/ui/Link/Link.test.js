import React from 'react';
import renderer from 'react-test-renderer';

import Link from './Link';


it('<Link /> renders correctly', () => {
  const tree = renderer
    .create(<Link>Lorem ipsum</Link>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Link className="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Link className="foo-bar">Lorem ipsum</Link>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Link href="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Link href="/">Lorem ipsum</Link>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
