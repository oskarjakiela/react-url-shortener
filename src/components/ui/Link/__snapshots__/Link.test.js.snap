// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly 1`] = `
<a
  className="Shorty-Link"
>
  Lorem ipsum
</a>
`;

exports[`<Link className="foo-bar" /> renders correctly 1`] = `
<a
  className="Shorty-Link foo-bar"
>
  Lorem ipsum
</a>
`;

exports[`<Link href="foo-bar" /> renders correctly 1`] = `
<a
  className="Shorty-Link"
  href="/"
>
  Lorem ipsum
</a>
`;
